middleware/faq.rst

```diff
@@ -535,6 +535,251 @@ The resulting query logs can be useful for developers and database administrator
+
+Statuses for Datasets
+^^^^^^^^^^^^^^^^^^^^^
+The statuses for datasets all pertain to whether a ``datasetType`` associated with a certain data ID exists in the `Butler`, and whether it could be used as an input to downstream tasks.
+
+Visible
+"""""""
+The dataset exists, and comes up in a find-first search (``Registry.queryDatasets``). This means it will be used as an input for downstream tasks.
+
+Shadowed
+""""""""
+The dataset exists in the `Butler`, but does not come up in a find-first search (``Registry.queryDatasets``). Therefore, we don't have to worry about it being used as an input.
+
+Predicted Only
+""""""""""""""
+This status occurs when the graph predicts an output dataset that ultimately is not produced by the Science Pipelines because it was deemed unnecessary when the graph was executed. These are commonly referred to as "NoWorkFound cases" because their output messages say "No work found for task" (and hence said task does not produce output datasets).
+
+Unsuccessful
+""""""""""""
+An unsuccessful status for a dataset just means it does not exist in the `Butler`. ``Unsuccessful`` datasets are the results of ``Failed`` and ``Blocked`` quanta.
+
+Cursed
+""""""
+A ``Cursed`` dataset is the result of an unsuccessful quantum which would otherwise be marked as ``Visible``. This means that if any datasets which were produced by non-``Successful`` quanta (this includes ``Unknown``, ``Failed``, ``Blocked`` and ``Wonky`` quanta) come up in a ``Registry.queryDatasets`` find-first search, we flag them as ``Cursed`` in order to halt processing, start investigation, and prevent them from being used as inputs later. Ideally ``Cursed`` datasets should be removed before processing continues.
+
+Total
+"""""
+The sum of all the previous categories.
+
+Expected
+""""""""
+The expected number of datasets, according to the graphs.
+
+As a final note, the number in ``Total`` should always match the number in ``Expected``. If it does not, this is a problem with this algorithm and should be reported to the developers.
+
+Using pipetask aggregate-reports to combine group-level pipetask report summaries into a step-level rollup
+----------------------------------------------------------------------------------------------------------
+`pipetask report` works on the group-level (on attempts to execute the same pipeline with the same data-query). However, it is possible to collate the JSON output from multiple groups into one file. This is intended for combining group-level summaries into summaries over processing steps, and answering questions like "What are all the errors that occurred in step 1?" or "How many quanta were blocked in step 5?"
```

Comment:
There needs to be an empty line between the header underline and the first sentence here.